activerecord/lib/active_record/counter_cache.rb

```diff
@@ -180,14 +181,24 @@ def decrement_counter(counter_name, id, by: 1, touch: nil)
       def counter_cache_column?(name) # :nodoc:
         _counter_cache_columns.include?(name)
       end
+
+      def load_schema # :nodoc:
```

Comment:
At some point I was cleaning up the list on `reload_schema_from_cache` but I think it's unnecessary since it's extremely unlikely application will drop the column and also remove the association in the runtime. It's also okay if we attempt to build the list twice as we maintaining uniqueness of the values through `list |= [new_value]`